### PR TITLE
feat: add plugin support

### PR DESCRIPTION
**Pull request by louis-bompart**
## Proposed changes

- `rm  packages/cli packages/cli/core`
- add `@oclif/plugins-plugin` (with a twist to be able to customize our prefix)

## Testing

CI YOLO

### CODE REVIEW

File: packages/cli/core/.prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const base = require('../../../.prettierrc');
```
Comment (louis-bompart) [inline packages/cli/core/.prettierrc.js:1]: Just changed the path

File: packages/cli/core/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "../../../base.eslintrc"
```
Comment (louis-bompart) [inline packages/cli/core/.eslintrc:2]: Just changed the path

File: packages/cli/core/project.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "root": "packages/cli/core",
+  "sourceRoot": "packages/cli/core/src",
```
Comment (louis-bompart) [inline packages/cli/core/project.json:3]: Just changed those paths

File: packages/cli-e2e/setup/utils.ts [hunk 4]
```diff
@@ -76,7 +76,7 @@ export function setProcessEnv() {
 }
 
 export function setCliExecPath() {
-  process.env.CLI_EXEC_PATH = resolve(__dirname, '../../cli/bin/dev');
+  process.env.CLI_EXEC_PATH = require.resolve('@coveo/cli/bin/dev');
```
Comment (louis-bompart) [inline packages/cli-e2e/setup/utils.ts:79]: ShouldWork™️ 

No relative path is gud path

File: README.md [hunk 5]
```diff
@@ -86,7 +86,7 @@ npm i
 To run the local version of the CLI
 
 ```sh
-./packages/cli/bin/run
+./packages/cli/core/bin/dev
```
Comment (y-lakhdar) [inline README.md:89]: good catch for the `dev`

File: packages/cli-e2e/setup/utils.ts [hunk 6]
```diff
@@ -76,7 +76,7 @@ export function setProcessEnv() {
 }
 
 export function setCliExecPath() {
-  process.env.CLI_EXEC_PATH = resolve(__dirname, '../../cli/bin/dev');
+  process.env.CLI_EXEC_PATH = resolve(__dirname, '../../cli/core//bin/dev');
```
Comment (y-lakhdar) [inline packages/cli-e2e/setup/utils.ts:79]:
```suggestion
  process.env.CLI_EXEC_PATH = resolve(__dirname, '../../cli/core/bin/dev');
```

File: packages/cli/core/package.json [hunk 7]
```diff
@@ -116,6 +118,8 @@
         "-v"
       ]
     },
+    "scope": "coveo",
+    "prefix": "cli-plugin",
```
Comment (y-lakhdar) [inline packages/cli/core/package.json:122]: What is that parameter doing?

File: patches/@oclif+plugin-plugins+2.1.0.patch [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+         const unfriendly = this.unfriendlyName(name);
+@@ -196,9 +197,10 @@ class Plugins {
+     }
+     friendlyName(name) {
+         const scope = this.config.pjson.oclif.scope;
++        const prefix = this.config.pjson.oclif.prefix ?? 'plugin';
+         if (!scope)
+             return name;
+-        const match = name.match(`@${scope}/plugin-(.+)`);
++        const match = name.match(`@${scope}/${prefix}-(.+)`);
```
Comment (y-lakhdar) [inline patches/@oclif+plugin-plugins+2.1.0.patch:25]: Nice. nvm my comment above.

**Review by olamothe — APPROVED** [review]

**Pull request merged**